README.md

```diff
@@ -1,30 +1,127 @@
 # WIP: Cerializer
-Cerializer aims at providing a even faster alternative to FastAvro and Avro standard library.
+Cerializer is an Avro de/serialization library that aims at providing a even faster alternative to FastAvro and Avro standard library.
 
-This speed increase does not come without a cost. Cerializer will work only with predefined set of schemata for which it will generate tailor made code. This way, the overhead caused by the universality of other serialization libraries will be avoided.
+This speed increase does not come without a cost. Cerializer will work only with predefined set of schemata for which it will generate tailor made Cython code. This way, the overhead caused by the universality of other serialization libraries will be avoided.
 
 Special credit needs to be given to [FastAvro](https://github.com/fastavro/fastavro) library, by which is this project heavily inspired.
 
+**Example of a schema and the corresponding code**
+
+SCHEMA
+```
+{
+    'name': 'array_schema', 
+    'doc': 'Array schema', 
+    'namespace': 'messaging', 
+    'type': 'record', 
+    'fields': [
+        {
+            'name': 'order_id', 
+            'doc': 'Id of order', 
+            'type': 'string'
+        }, 
+        {
+            'name': 'trades', 
+            'type': {
+                'type': 'array', 
+                'items': ['string', 'int']
+            }
+        }
+    ]
+}
+```
+
+CORRESPONDING CODE
+```
+def serialize(data, output):
+    cdef bytearray buffer = bytearray()
+    cdef dict datum
+    cdef str type_51
+
+    write.write_string(buffer, data['order_id'])
+
+    if len(data['trades']) > 0:
+        write.write_long(buffer, len(data['trades']))
+        for val_59 in data['trades']:
+            if type(val_59) is tuple:
+                type_51, val_60 = val_59
+                
+                if type_51 == 'string':
+                    write.write_long(buffer, 0)
+                    write.write_string(buffer, val_60)
+                
+                elif type_51 == 'int':
+                    write.write_long(buffer, 1)
+                    write.write_int(buffer, val_60)
+                
+
+            else:
+
+                if type(val_59) is str:
+                    write.write_long(buffer, 0)
+                    write.write_string(buffer, val_59)
+
+                elif type(val_59) is int:
+                    write.write_long(buffer, 1)
+                    write.write_int(buffer, val_59)
+
+    write.write_long(buffer, 0)
+    output.write(buffer)
+
+
+
+def deserialize(fo):
+    cdef long long i_62
+    cdef long long i_63
+    cdef long i_64
+    data = {}
+    data['order_id'] = read.read_string(fo)
+    data['trades'] = []
+
+    i_63 = read.read_long(fo)
+    while i_63 != 0:
+        if i_63 < 0:
```

Comment:
How can be the number of elements in array negative?

Reply:
> """Arrays are encoded as a series of blocks.
    Each block consists of a long count value, followed by that many array
    items.  A block with count zero indicates the end of the array.  Each item
    is encoded per the array's item schema.
    If a block's count is negative, then the count is followed immediately by a
    long block size, indicating the number of bytes in the block.  The actual
    count in this case is the absolute value of the count written.  """

https://github.com/fastavro/fastavro/blob/master/fastavro/_write.pyx#L158